problem/src/main/java/org/zalando/problem/ThrowableProblem.java

```diff
@@ -22,9 +22,29 @@
 
 import javax.annotation.concurrent.Immutable;
 
+import static com.google.common.base.Preconditions.checkState;
+
 @Immutable
 public abstract class ThrowableProblem extends RuntimeException implements Problem {
 
-    // TODO support ThrowableProblem getCause();
+    @Deprecated
+    public ThrowableProblem() {
+        
+    }
+
+    public ThrowableProblem(Throwable cause) {
+        super(cause);
+    }
+
+    @Override
+    public ThrowableProblem getCause() {
+        final Throwable cause = super.getCause();
+        if (cause == null) {
+            return null;
+        }
 
+        checkState(cause instanceof ThrowableProblem, "Expected throwable problem cause, but got %s", cause);
```

Comment:
Can we check the cause in the constructor instead? That would give a better stacktrace to find the error.